eng/common/Update-Change-Log.ps1

```diff
@@ -0,0 +1,125 @@
+param (
+  [Parameter(Mandatory = $true)]
+  [String]$Version,
+  [Parameter(Mandatory = $true)]
+  [String]$ChangeLogPath,
+  [String]$Unreleased = $True,
+  [String]$ReplaceVersion = $False
+)
+
+# Parameter description
+# Version : Version to add or replace in change log
+# ChangeLogPath: Path to change log file. If change log path is set to directory then script will probe for change log file in that path 
+# Unreleased: Default is true. If it is set to false, then today's date will be set in verion title. If it is True then title will show "Unreleased"
+# ReplaceVersion: This is useful when replacing current version title with new title.( Helpful to update the title before package release)
+
+
+function Get-ChangelogPath
+{
+  Param (
+        [String]$Path
+   )
+  
+   # Check if CHANGELOG.md is present in path 
+   $ChangeLogPath = Join-Path -Path $Path -ChildPath "CHANGELOG.md"
+   if ((Test-Path -Path $ChangeLogPath) -eq $False){
+      # Check if change log exists with name HISTORY.md
+      $ChangeLogPath = Join-Path -Path $Path -ChildPath "HISTORY.md"
+      if ((Test-Path -Path $ChangeLogPath) -eq $False){
+         Write-Host "Change log is not found in path[$Path]"
+         exit(1)
+      }
+   }
+  
+   Write-Host "Change log is found at path [$ChangeLogPath]"
+   return $ChangeLogPath
+}
+
+
+function Get-VersionTitle
+{
+   Param (
+        [String]$Version,
+        [String]$Unreleased
+   )
+   # Generate version title
+   $newVersionTitle = "## $Version Unreleased" -join [Environment]::NewLine
+   if ($Unreleased -eq $False){
+      $releaseDate = Get-Date -Format "(yyyy-MM-dd)"
+      $newVersionTitle = "## $Version $releaseDate" -join [Environment]::NewLine      
+   }
+   return $newVersionTitle
+}
+
+
+function Get-NewChangeLog
+{
+   Param (
+        [System.Collections.ArrayList]$ChangelogLines,
+        [String]$Version,
+        [String]$Unreleased,
+        [String]$ReplaceVersion
+   ) 
+
+   # Version Parser module
+   Import-Module $PSScriptRoot/Version-Parser.psm1
+
+   # version parameter is to pass new version to add or replace
+   # Unreleased parameter can be set to False to set today's date instead of "Unreleased in title"
+   # ReplaceVersion param can be set to true to replace current version title( useful at release time to change title)
+
+   # find index of current version
+   $Index = 0
+   $CurrentTitle = ""
+   for(; $Index -lt $ChangelogLines.Count; $Index++){
+      if (Version-Matches($ChangelogLines[$Index])){
+         $CurrentTitle = $ChangelogLines[$Index]
+         Write-Host "Current Version title: $CurrentTitle"
+         break
+      }
+   }
+
+   # Generate version title
+   $newVersionTitle = Get-VersionTitle -Version $Version -Unreleased $Unreleased
+
+   # if version is already found and not replacing then nothing to do
+   if ($ReplaceVersion -eq $False){
+      # Check if version is already present in log      
+      if ($CurrentTitle.Contains($Version)){
```

Comment:
Don't you want to check the unreleased part as well? I'd suggest you check to see if the current title exactly matches the newVersionTitle and if it does then nothing to do but if not they set it to that.

Reply:
In fact if you want you can blindly set the string all the time and if nothing changed then there shouldn't be any changes to the file but it doesn't hurt to write it still.

Reply:
Ignore my second comment you shouldn't always blindly update because in a lot of cases you need to insert a new version. However if the version does match you should still check to see if the unreleased state also matches and if not update that part at least.